src/ValueFormatters/QuantityFormatter.php

```diff
@@ -197,16 +226,22 @@ private function getRoundingExponent( QuantityValue $quantity ) {
 	 * @return string|null Text
 	 */
 	private function formatMargin( DecimalValue $margin, $roundingExponent ) {
-		if ( $this->options->getOption( self::OPT_SHOW_UNCERTAINTY_MARGIN ) ) {
-			// TODO: never round to 0! See bug #56892
-			$roundedMargin = $this->decimalMath->roundToExponent( $margin, $roundingExponent );
+		$marginModel = $this->options->getOption( self::OPT_SHOW_UNCERTAINTY_MARGIN );
 
-			if ( !$roundedMargin->isZero() ) {
-				return $this->decimalFormatter->format( $roundedMargin );
-			}
+		if ( $marginModel === self::SHOW_UNCERTAINTY_MARGIN_NEVER ) {
+			return null;
+		}
+
+		// TODO: never round to 0! See bug #56892
+		$roundedMargin = $this->decimalMath->roundToExponent( $margin, $roundingExponent );
```

Comment:
Giving a margin of uncertainty with more precision than the value itself is also never correct.
But I agree that we should just show this as entered, correct or not.